src/test/java/sk/uniba/fmph/dcs/game_phase_controller/AllPlayersTakeARewardStateTest.java

```diff
@@ -0,0 +1,62 @@
+package sk.uniba.fmph.dcs.game_phase_controller;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import sk.uniba.fmph.dcs.stone_age.ActionResult;
+import sk.uniba.fmph.dcs.stone_age.Effect;
+import sk.uniba.fmph.dcs.stone_age.HasAction;
+import sk.uniba.fmph.dcs.stone_age.InterfaceTakeReward;
+import sk.uniba.fmph.dcs.stone_age.PlayerOrder;
+
+public class AllPlayersTakeARewardStateTest {
+
+    private static class MockTakeReward implements InterfaceTakeReward {
+        private boolean returnValue;
+        private HasAction hasAction;
+
+        public MockTakeReward(boolean returnValue, HasAction hasAction) {
+            this.returnValue = returnValue;
+            this.hasAction = hasAction;
+        }
+
+        @Override
+        public Boolean takeReward(PlayerOrder player, Effect reward) {
+            return returnValue;
+        }
+
+        @Override
+        public HasAction tryMakeAction(PlayerOrder player) {
+            return hasAction;
+        }
+    }
+
+    @Test
+    public void testMakeAllPlayersTakeARewardChoice() {
+        PlayerOrder player = new PlayerOrder(1, 1);
+
+        // Test case: Reward is successfully taken
```

Comment:
Jeden test by mal testovať jednu vec (jeden scenar). Keď potom failne tak je jasné z akeho dôvodu. V takomto prípade pri failnuti nevieš či je problem pri úspešnom, neuspešnom scenary alebo oboch.

Fajn, že si aspon pridal komentare aky scenar daná časť testuje, no lepšie je to oddeliť do samostatnch metód. To čo sa má stať je pak jasné hneď z nazvu metody, pri failnuti ti junit vypíše v akych scenaroch je problem a druhý scenar sa vykona aj v pripade, že failne prvý.